### ISSUE TITLE
Issues related to week 2

### ISSUE THREAD

**Issue by aksh1618** [chay2199/GitHeat#2]
Appreciable effort !! However there are some faults in your approach:

- ***A lot of redundancy in code***. Say you just have to remove those 4 'hash-lines' (why are there 4 anyway? 1 should be enough :upside_down_face: ) or change prompt text, you'd have to do it at multiple places. Basically, anything you copy pasted should belong to a function instead. So refactor your code to reuse parts.

- ***No separation of logic and io***. You should separate the parts which the user interacts with from the parts which do the actual work (calculating directory size etc.) into different modules (files). This will help in easier maintenance of the project.

-  ***Useless pie chart***. It is only useful if I'm using it for the root of the partition. But for any other directory, it is useless, as it shows the total size of this folder with the free space of the whole directory. What we want is to see what files are taking what percentage of the folder size, not the folder size itself. This should be easily modifiable, once you refactor and restructure your code.

See this screenshot (I expect you to at least implement the innermost circle of the right part):

![deepinscreenshot_20180614193756](https://user-images.githubusercontent.com/26065812/41417301-895b3130-700a-11e8-991c-d7243dae9057.png)

- ***Other bugs***. 
  - If I press -3, it takes me to the root on linux instead of parent directory,
  - If I enter index of a file, it shows warning but crashes when I enter another index.:
```
0
You are currently in  /run/media/aksh/AKSH/DC/op2bck/Facebook/FB_IMG_1488564715037.jpg/
#####################################
#####################################
#####################################
#####################################
#####################################
These are all the files and folders in this directory: 
Oops the index you have selected is of a file. You can try again!!
Your file size in MB:  0.035526
Enter -1 to kill the program or -2 to know the details about that folder or the index number to further enter in a directory.
1
Traceback (most recent call last):
  File "DiskAnalyser.py", line 231, in <module>
    pathName = pathName + str(dirlist[choice]) + '/'
IndexError: list index out of range
```

Basically, I think this has a really good base with good potential, but what it does right now is something that a file manager on any OS is already capable of doing. The first thing you need to do is stop using a notebook and start using an editor / IDE for programming. Then refactor and restructure your code, and add sizes of children folders to the pie chart, and we could have a *really* good and useful little utility :slightly_smiling_face:

**Comment by chay2199** [comment]
Thanks for the valuable input. I will try my level best to add more features and fix the bugs as soon as possible.
